### ISSUE TITLE
OpenAPI 3 Model seems to be using an OpenAPI 2 SecurityRequirements structure

### ISSUE THREAD

**Issue by TristanSpeakEasy** [pb33f/libopenapi#20]
From my understanding of the OpenAPI 3 spec, the `security` field of a document or operation is an array of maps ie `[]map[string][]string` but the `Security` field of an v3.Document is `*base.SecurityRequirement` which contains a field `Requirements map[string][]string` so is not capturing that fact its an array of security requirements.

For reference https://swagger.io/docs/specification/authentication/

**Comment by daveshanley** [comment]
You're correct. The `Document` needs to capture `[]*base.SecurityRequirement` and not a single `*base.SecurityRequirement`

I will work on a fix for this soon. It's likely to be a breaking change, which means a feature version bump to `v.0.3`

**Comment by daveshanley** [comment]
This is actually a good find; after some analysis, I discovered:

- v2 models are correct
- v3 low '`Document` model is correct
- v3 high `Document` model is incorrect
- v3 high model ignores `security` completely! (I forgot to add it to the high model)

So, thank you, there was no test or check for this. I am working on a fix.

**Comment by TristanSpeakEasy** [comment]
Thanks for the fast turn around!
